### PR TITLE
chore(tests): unmochaify transaction unit tests

### PR DESCRIPTION
**Pull request by jordigh**
## Description

* Closes NEWRELIC-8839.

### CODE REVIEW

File: test/unit/transaction.test.js [hunk 1]
```diff
@@ -1285,7 +1367,7 @@ describe('Transaction', function () {
           ty: 'Mobile',
           ac: trustedAccountKey,
           ap: '51424',
-          id: '5f474d64b9cc9b2a',
+          id: '5f4n74d64b9cc9b2a',
```
Comment (mrickard) [inline test/unit/transaction.test.js:1370]: It looks like this id value has changed.

Comment (jordigh) [reply test/unit/transaction.test.js:1370]: Whoa! I know how that happened. Emacs-induced typo. :( `C-a` means "beginning of line" so sometimes I typo and leave a stray "a" somewhere.

Comment (mrickard) [reply test/unit/transaction.test.js:1370]: Though this one looks like an inserted `n`, I'd think.

Comment (jordigh) [reply test/unit/transaction.test.js:1370]: `C-n` is next line. :P (down arrow)

Comment (mrickard) [reply test/unit/transaction.test.js:1370]: Ah, that'd explain it, yeah.

Comment (mrickard) [inline test/unit/transaction.test.js:1370]:
```suggestion
          id: '5f474d64b9cc9b2a',
```

File: test/unit/transaction.test.js [hunk 2]
```diff
@@ -23,528 +16,534 @@ const Segment = require('../../lib/transaction/trace/segment')
...
   })
+})
 
-  it('should know when it is not a web transaction', function () {
-    const tx = new Transaction(agent)
-    tx.type = Transaction.TYPES.BG
-    expect(tx.isWeb()).to.be.false
+tap.test('Transaction naming tests', (t) => {
+  t.autoend()
+  const agent = helper.loadMockedAgent({
```
Comment (mrickard) [inline test/unit/transaction.test.js:211]: Previously, this test group used a new agent for each subtest. Is it important to preserve that?

File: test/unit/transaction.test.js [hunk 3]
```diff
@@ -23,528 +16,534 @@ const Segment = require('../../lib/transaction/trace/segment')
...
+      txn.finalizeNameFromUri(url, 200)
+      t.equal(txn.name, 'WebTransaction/NormalizedUri/test/*/string/*')
+      t.end()
+    })
 
-      it('produces the custom name even when status is 501', function () {
-        trans.finalizeNameFromUri('/test/string?do=thing&another=thing', 501)
-        expect(trans.name).equal('WebTransaction/Custom/test')
-      })
+    t.test('should not scope web transactions to their URL', (t) => {
```
Comment (mrickard) [inline test/unit/transaction.test.js:440]: Is this subtest new for this group?

Comment (jordigh) [reply test/unit/transaction.test.js:440]: Yes, but I think it belongs with this group. Originally it was alone at the "top level". We are checking that its name isn't scoped. In particular, when a partial name isn't set.

File: test/unit/transaction.test.js (hunk 2 above)
Comment (jordigh) [reply test/unit/transaction.test.js]: Okay, I've moved the tear-down/setup logic closer to each subtest to ensure the agent gets thrown out every time.

File: test/unit/transaction.test.js [hunk 4]
```diff
@@ -23,1251 +16,1326 @@ const Segment = require('../../lib/transaction/trace/segment')
...
-        txn.end()
-      })
-    })
+    t.end()
   })
 
-  it('should hand itself off to the agent upon finalization', function (done) {
-    agent.on('transactionFinished', function (inner) {
-      expect(inner).equal(trans)
+  t.test('with distributed tracing enabled', (t) => {
```
Comment (mrickard) [inline test/unit/transaction.test.js:74]: Is this the same as the block on line 59?

Comment (jordigh) [reply test/unit/transaction.test.js]: It's not, on line 59 we are not ignoring the transaction. I've relabelled the tests to make this clearer.

File: test/unit/transaction.test.js [hunk 5]
```diff
@@ -71,11 +71,11 @@ tap.test('Transaction unit tests', (t) => {
     t.end()
   })
 
-  t.test('with distributed tracing enabled', (t) => {
+  t.test('with DT enabled, should not produce span events when ignored', (t) => {
```
Comment (mrickard) [inline test/unit/transaction.test.js:74]: Ah, this is much clearer--thank you!

**Review by mrickard — APPROVED** [review]

**Pull request merged**